sw/device/sca/aes_serial/prng.c

```diff
@@ -0,0 +1,189 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stddef.h>
+#include <stdint.h>
+
+// TODO(alphan): Using MT for now as a proof of concept to minimize changes on
+// the host side. We should probably replace this with a PRNG from xoshiro*
+// family for PRNGs, e.g. xoshiro128++, for better performance and less
+// overhead.
+
+/**
+ * Mersenne Twister PRNG modified from:
+ * http://www.math.sci.hiroshima-u.ac.jp/m-mat/MT/MT2002/CODES/mt19937ar.c
+ *
+ * The code below is identical to the code at the link above except for:
+ *    - Unused functions were removed,
+ *    - `static` speficier was added to all remaining functions,
+ *    - Data types were replaced with compatible types for consistency, and
+ *    - `clang-format` was run.
+ *
+ * Verbatim comments from original code:
+ *
+ *   A C-program for MT19937, with initialization improved 2002/1/26.
+ *   Coded by Takuji Nishimura and Makoto Matsumoto.
+ *
+ *   Before using, initialize the state by using init_genrand(seed)
+ *   or init_by_array(init_key, key_length).
+ *
+ *   Copyright (C) 1997 - 2002, Makoto Matsumoto and Takuji Nishimura,
```

Comment:
I think you have to add this copyright notice to the very top and then define an exclusion for the CI license checker. See for example `hw/ip/usbdev/rtl/usbdev_iomux.sv`.

Reply:
Done, thanks!